cmd/oras/root/blob/fetch.go

```diff
@@ -147,40 +147,35 @@ func (opts *fetchBlobOptions) doFetch(ctx context.Context, src oras.ReadOnlyTarg
 		vr := content.NewVerifyReader(rc, desc)
 
 		// outputs blob content if "--output -" is used
-		if opts.outputPath == "-" {
-			if _, err := io.Copy(os.Stdout, vr); err != nil {
-				return ocispec.Descriptor{}, err
-			}
-			if err := vr.Verify(); err != nil {
+		writer := os.Stdout
+		if opts.outputPath != "-" {
+			// save blob content into the local file if the output path is provided
+			file, err := os.Create(opts.outputPath)
+			if err != nil {
 				return ocispec.Descriptor{}, err
 			}
-			return desc, nil
-		}
-
-		// save blob content into the local file if the output path is provided
-		file, err := os.Create(opts.outputPath)
-		if err != nil {
-			return ocispec.Descriptor{}, err
+			writer = file
+			defer func() {
+				if err := file.Close(); fetchErr == nil {
+					fetchErr = err
+				}
+			}()
```

Comment:
nit: coding style: `defer` first after error handling then other code
```suggestion
			defer func() {
				if err := file.Close(); fetchErr == nil {
					fetchErr = err
				}
			}()
			writer = file
```